library/src/main/java/com/subsub/library/BeautyButton.kt

```diff
@@ -75,15 +76,23 @@ class BeautyButton : AppCompatTextView {
 
         gravity = Gravity.CENTER
 
-        topLeftCornerRect = Rect(0, 0, customShadowRadius.toInt(), customShadowRadius.toInt())
-        topRect = Rect(0, 0, (width - customShadowRadius * 2 + myElevation * 2 - ((1 - shadowWidth) * width)).toInt(), customShadowRadius.toInt())
-        topRightCornerRect = Rect(0, 0, customShadowRadius.toInt(), customShadowRadius.toInt())
-        rightRect = Rect(0, 0, customShadowRadius.toInt(), (height - customShadowRadius * 2 + myElevation * 2 - (1 - shadowHeight) * height).toInt())
-        bottomRightCornerRect = Rect(0, 0, customShadowRadius.toInt(), customShadowRadius.toInt())
-        bottomRect = Rect(0, 0, (width - customShadowRadius * 2 + myElevation * 2 - ((1 - shadowWidth) * width)).toInt(), customShadowRadius.toInt())
-        bottomLeftCornerRect = Rect(0, 0, customShadowRadius.toInt(), customShadowRadius.toInt())
-        leftRect = Rect(0, 0, customShadowRadius.toInt(), (height - customShadowRadius * 2 + myElevation * 2 - (1 - shadowHeight) * height).toInt())
+        with(customShadowRadius) {
+            topLeftCornerRect = Rect(0, 0, toInt(), toInt())
+            topRect = Rect(0, 0, (width - customShadowRadius * 2 + myElevation * 2
+                    - ((1 - shadowWidth) * width)).toInt(), toInt())
 
+            topRightCornerRect = Rect(0, 0, toInt(), toInt())
+            rightRect = Rect(0, 0, toInt(), (height - customShadowRadius * 2 + myElevation * 2
+                    - (1 - shadowHeight) * height).toInt())
+
+            bottomRightCornerRect = Rect(0, 0, toInt(), toInt())
+            bottomRect = Rect(0, 0, (width - customShadowRadius * 2 + myElevation * 2
+                    - ((1 - shadowWidth) * width)).toInt(), toInt())
+
+            bottomLeftCornerRect = Rect(0, 0, toInt(), toInt())
+            leftRect = Rect(0, 0, toInt(), (height - customShadowRadius * 2 + myElevation * 2
+                    - (1 - shadowHeight) * height).toInt())
+        }
```

Comment:
@antoxa2584x Really nice! I didn't know there's `with` function 😄 
But there're still can be improved though, all `customShadowRadius` can be replaced with `this` I suppose...